website/pages/en/cookbook/how-to-secure-api-keys-using-nextjs-server-components.mdx

```diff
@@ -0,0 +1,125 @@
+---
+title: How to Secure API Keys Using Next.js Server Components
+---
+
+## Overview
+
+We can use [NextJS server components](https://nextjs.org/docs/app/building-your-application/rendering/server-components) to properly secure our API key from exposure in the front-end of our dapp. To further increase our API key security, we can also [restrict our API key to certain subgraphs or domains in Subgraph Studio](/cookbook/upgrading-a-subgraph/#securing-your-api-key).
+
+In this cookbook, we will go over how to create a NextJS server component that queries a subgraph while also hiding the API key from the front-end.
+
+### Caveats
+
+- NextJS server components do not protect API keys from being drained using denial of service attacks.
+- The Graph Network gateways have denial of service detection and mitigation strategies in place, however using server components may weaken these protections.
+- NextJS server components introduce centralization risks as the server can go down.
+
+### Why It's Needed
+
+In a standard React application, API keys included in the front-end code can be exposed to the client-side, posing a security risk. While `.env` files are commonly used, they don't fully protect the keys since React's code is executed on the client side, exposing the API key in the headers. Next.js Server Components address this issue by handling sensitive operations server-side.
+
+### Using client-side rendering to query a subgraph
+
+![Client-side rendering](/img/api-key-client-side-rendering.png)
+
+### Prerequisites
+
+- An API key from [www.thegraph.com/studio](www.thegraph.com/studio)
+- Basic knowledge of Next.js and React.
+- An existing Next.js project.
```

Comment:
Many people can be using `pages/` and this won't be supported. So we should be explicit and can also link next docs https://nextjs.org/docs/app/building-your-application/rendering/server-components
```suggestion
- An existing Next.js project using app router
```

Reply:
How about this:

```suggestion
- An existing Next.js project that uses the [Next.js App Router](https://nextjs.org/docs/app).
```